sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/ModuleScanContainer.java

```diff
@@ -78,31 +75,29 @@
 
 public class ModuleScanContainer extends ComponentContainer {
   private static final Logger LOG = LoggerFactory.getLogger(ModuleScanContainer.class);
-  private final Project module;
+  private final DefaultInputModule module;
 
-  public ModuleScanContainer(ProjectScanContainer parent, Project module) {
+  public ModuleScanContainer(ProjectScanContainer parent, DefaultInputModule module) {
     super(parent);
     this.module = module;
   }
 
   @Override
   protected void doBeforeStart() {
-    LOG.info("-------------  Scan {}", module.getName());
+    LOG.info("-------------  Scan {}", module.definition().getName());
     addCoreComponents();
     addExtensions();
   }
 
   private void addCoreComponents() {
-    ProjectDefinition moduleDefinition = getComponentByType(DefaultProjectTree.class).getProjectDefinition(module);
     add(
-      moduleDefinition,
+      module.definition(),
       module,
-      getComponentByType(BatchComponentCache.class).get(module).inputComponent(),
       ModuleSettings.class);
 
     // hack to initialize settings before ExtensionProviders
     ModuleSettings moduleSettings = getComponentByType(ModuleSettings.class);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "moduleSettings". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused "moduleSettings" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1481)